pkg/rtc/room.go

```diff
@@ -362,24 +371,34 @@ func (r *Room) UpdateSubscriptions(
 		for _, trackSid := range pt.TrackSids {
 			track := p.GetPublishedTrack(trackSid)
 			if track != nil {
-				tracks[trackSid] = track
+				trackInfos[trackSid] = struct {
+					publisher types.Participant
+					track     types.PublishedTrack
+				}{
+					publisher: p,
+					track:     track,
+				}
 			}
 		}
 	}
 
 	// handle subscription changes
-	for _, track := range tracks {
+	for trackSid, trackInfo := range trackInfos {
 		if subscribe {
-			if err := track.AddSubscriber(participant); err != nil {
+			if _, err := trackInfo.publisher.AddSubscriber(participant, types.AddSubscriberParams{TrackSids: []string{trackSid}}); err != nil {
 				return err
 			}
 		} else {
-			track.RemoveSubscriber(participant.ID())
+			trackInfo.track.RemoveSubscriber(participant.ID())
```

Comment:
This is kind of ugly. `AddSubscriber` goes through the publishing participant. But, `RemoveSubscriber` is going directly to `track`. Did this to avoid adding another method to `Participant` interface. I think it is worth adding a method to `Participant` interface for a few reasons
- Symmetry
- Can make this code simpler, i. e. can get rid of the ad hoc struct here
- Currently, I am not removing the subscriber from pending subscriptions (it gets handled when processing the next update of subscription permissions). If `RemoveSubscriber` goes through `Participant`, can update the pending subscription to remove this subscriber.

Reply:
RemoveSubscriber to  Participant  make sense. and why there are two parameters trackIds and participantTrackIds, because compatible?

Reply:
Missed one case here. A subscriber participant leaving is not covered though. It is all triggered from down track close. Guess, I need to have that flow back to the publisher participant and handle updating pending subscriptions and permissions too.

But, still `RemoveSubscriber` on `Participant` seems nicer to have.